plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -0,0 +1,247 @@
+
+        if key not in data:
+            return {'ret': False, 'msg': "Key %s not found" % key}
+
+        TimeZonelist = data["TimeZoneList"]
+        index = ""
+
+        for i in range(0, len(TimeZonelist)):
+            if(attr['mgr_attr_value'] in TimeZonelist[i]["Name"]):
+                index = TimeZonelist[i]["Index"]
+                break
+
+        payload = {key: {"Index": index}}
+        # toset = "{\"Index\":" + str(index) + "}"
+
+        # payload = {key: json.loads(toset)}
+
+        response = self.patch_request(self.root_uri + uri, payload)
+        if(response['ret'] is False):
+            return response
+
+        return {'ret': True, 'changed': True, 'msg': "Modified %s" % attr['mgr_attr_name']}
+
+    def set_DNSserver(self, attr):
+        # result = {}
+        key = attr['mgr_attr_name']
+        # setkey = "Oem"
+
+        nic_info = self.get_manager_ethernet_uri()
+        uri = nic_info["nic_addr"]
+
+        response = self.get_request(self.root_uri + uri)
+        if(response['ret'] is False):
+            return response
+
+        data = response['data']
+
+        DNSlist = data["Oem"]["Hpe"]["IPv4"][key]
+
+        if(len(DNSlist) == 3):
+            DNSlist.pop(0)
+
+        DNSlist.append(attr['mgr_attr_value'])
+
+        payload = {
+            "Oem": {
+                "Hpe": {
+                    "IPv4": {
+                        key: DNSlist
+                    }
+                }
+            }
+        }
+
+        response = self.patch_request(self.root_uri + uri, payload)
+        if(response['ret'] is False):
+            return response
+
+        return {'ret': True, 'changed': True, 'msg': "Modified %s" % attr['mgr_attr_name']}
+
+    def set_DomainName(self, attr):
+        key = attr['mgr_attr_name']
+
+        nic_info = self.get_manager_ethernet_uri()
+        Ethuri = nic_info["nic_addr"]
+
+        response = self.get_request(self.root_uri + Ethuri)
+        if(response['ret'] is False):
+            return response
+
+        data = response['data']
+
+        payload = {"DHCPv4": {
+            "UseDomainName": ""
+        }}
+
+        if(data["DHCPv4"]["UseDomainName"] is True):
+            payload["DHCPv4"]["UseDomainName"] = False
+            resDHV4 = self.patch_request(self.root_uri + Ethuri, payload)
+            if resDHV4['ret'] is False:
+                return resDHV4
+
+        payload = {"DHCPv6": {
+            "UseDomainName": ""
+        }}
+
+        if(data["DHCPv6"]["UseDomainName"] is True):
+            payload["DHCPv6"]["UseDomainName"] = False
+            resDHV6 = self.patch_request(self.root_uri + Ethuri, payload)
+            if resDHV6['ret'] is False:
+                return resDHV6
+
+        domain_name = attr['mgr_attr_value']
+
+        payload = {}
+        payload = {"Oem": {
+            "Hpe": {
+                key: domain_name
+            }
+        }}
+
+        response = self.patch_request(self.root_uri + Ethuri, payload)
```

Comment:
The way that `payload` is set, and overwritten multiple times strikes me as odd, to say the least. I think this method should have at least a unit test for it.

And maybe - depending on the API - it could be improved to place only one `patch` request instead of multiple ones.